dstep/translator/Type.d

```diff
@@ -121,32 +121,114 @@ string translateSelector (string str, bool fullName = false, bool translateIdent
     return translateIdentifier ? .translateIdentifier(str) : str;
 }
 
+package bool isPrimitive(Type type)
+{
+    switch (type.kind)
+    {
+        case CXTypeKind.CXType_Void:
+        case CXTypeKind.CXType_Bool:
+        case CXTypeKind.CXType_Char_U:
+        case CXTypeKind.CXType_UChar:
+        case CXTypeKind.CXType_Char16:
+        case CXTypeKind.CXType_Char32:
+        case CXTypeKind.CXType_UShort:
+        case CXTypeKind.CXType_UInt:
+        case CXTypeKind.CXType_ULong:
+        case CXTypeKind.CXType_ULongLong:
+        case CXTypeKind.CXType_UInt128:
+        case CXTypeKind.CXType_Char_S:
+        case CXTypeKind.CXType_SChar:
+        case CXTypeKind.CXType_WChar:
+        case CXTypeKind.CXType_Short:
+        case CXTypeKind.CXType_Int:
+        case CXTypeKind.CXType_Long:
+        case CXTypeKind.CXType_LongLong:
+        case CXTypeKind.CXType_Int128:
+        case CXTypeKind.CXType_Float:
+        case CXTypeKind.CXType_Double:
+        case CXTypeKind.CXType_LongDouble:
+        case CXTypeKind.CXType_NullPtr:
+            return true;
+
+        default:
+            return false;
+    }
+}
+
 private:
 
-string translateTypedef (Context context, Type type)
-in
+bool isIdiomatic(Context context, Type type)
 {
-    assert(type.kind == CXTypeKind.CXType_Typedef);
+    switch (type.spelling)
+    {
+        case "NSInteger":
+        case "BOOL":
+        case "size_t":
+        case "ptrdiff_t":
+            return true;
+
+        default:
+            return false;
+    }
 }
-body
+
+string translateLongAlike(Context context, Type type)
+{
+    auto kind = type.canonical.kind;
+
+    if (kind == CXTypeKind.CXType_Long)
+    {
+        switch (type.spelling)
+        {
+            case "int64_t":
+            case "__s64":
```

Comment:
Is this standard/common typedef? Not sure if I like to add too much special cases.